### PR TITLE
Initial commit for ember-cli 2.15.0 support

### PR DESCRIPTION
**Pull request by jasonmit**

### CODE REVIEW

**Comment by jasonmit** [comment]
@SergeAstapov ready for review when you are ready.

File: index.js [hunk 1]
```diff
@@ -1,59 +1,59 @@
...
-    if (!configFilePath) {
-      configFilePath = path.join(project.root, '.env');
+  /**
+   * NOTE: dotenv needs to be invoked before the app config is materialized
+   * so that the process.env's are set appropriately.  Previously this was done
+   * within the `config` hook.  As of 2.15.x, that is too late in the process
+   * and needed to be moved into `init`.
+   */
+  init() {
+    this._super && this._super.apply(this, arguments);
```
Comment (SergeAstapov) [inline index.js]: @jasonmit this line seems to be failing on CI. Could you please change to `this._super.init && this._super.init.apply(this, arguments);` as per docs https://ember-cli.com/api/classes/Addon.html#method_init

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ember-cli-dotenv",
-  "version": "1.2.0",
+  "version": "2.0.0",
```
Comment (SergeAstapov) [inline package.json]: @jasonmit do you mind to keep it `1.2.0` for now and bump to 2.0 separately? I'd like to add FastBoot support in 2.0 as well

**Comment by jasonmit** [comment]
@SergeAstapov good feedback, I've addressed the concerns.  Thanks.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 [![npm version](https://badge.fury.io/js/ember-cli-dotenv.svg)](https://badge.fury.io/js/ember-cli-dotenv)
 [![Ember Observer Score](https://emberobserver.com/badges/ember-cli-dotenv.svg)](http://emberobserver.com/addons/ember-cli-dotenv)
+
 # Ember CLI Dotenv
 
 # Installation
 
 `ember install ember-cli-dotenv`
 
+# Upgrading to 2.0.0
```
Comment (jasonmit) [inline README.md]: Unsure how we want to deal with this, are you planning to land your fastboot work soon or should I defer the README changes until after we version?  @SergeAstapov

**Comment by jasonmit** [comment]
CI has been failing for awhile, we should address in a follow up PR.  Likely can be addressed when we upgrade ember-cli.

First time it started failing a few years ago with the error we are still seeing today: https://travis-ci.org/fivetanley/ember-cli-dotenv/jobs/116980605

File: README.md (hunk 3 above)
Comment (SergeAstapov) [reply README.md]: I do actually have a branch with FastBoot support, but it appeared there is a small issue in ember-cli-fastboot that blocks this addon to have FastBoot support, see https://github.com/ember-fastboot/ember-cli-fastboot/issues/543

To not block people from using this addon I think we can
* merge this PR
* merge PR with ember-cli update
* fix tests - I can do this tomorrow
* update readme and release 2.0
* add FastBoot support
* release 2.1

What about such plan?

Comment (jasonmit) [reply README.md]: Sounds good to me!

Comment (SergeAstapov) [reply README.md]: @jasonmit I'll take care of ember-cli upgrade (need to resolve conflict) and failing tests so we would be able to release version 2.0 shortly

**Pull request merged**